build-tests/sparo-outputs/src/generateOutputs.ts

```diff
@@ -0,0 +1,202 @@
+   *
+   * NOTE: please make sure the commands can be run in parallel
+   */
+  await FileSystem.ensureEmptyFolderAsync(tempFolder);
+  await Async.forEachAsync(
+    scenarios,
+    async (scenario: IScenarioDefinition) => {
+      const { name, args } = scenario;
+      const result: SpawnSyncReturns<string> = Executable.spawnSync(binPath, args);
+
+      if (result.status !== 0) {
+        throw new Error(
+          `Failed to run "sparo ${args.join(' ')}" with exit code ${result.status}\n${result.stderr}`
+        );
+      }
+
+      const outputPath: string = path.join(tempFolder, `${name}.txt`);
+      FileSystem.writeFile(
+        outputPath,
+        `Running "sparo ${args.join(' ')}":\n${processVersionString(result.stdout)}`
+      );
+    },
+    {
+      concurrency: 10
+    }
+  );
+
+  /**
+   * Files under outFolderPath are tracked by Git, files under inFolderPath are temporary files. During a local build,
+   * --production is false, temporary files are copied to outFolderPath. During a CI build, --production is true, the
+   * files with same name under these two folders are compared and CI build fails if they are different.
+   *
+   * This ensures that the temporary files must be up to date in the PR, and people who review the PR must approve any
+   * changes.
+   */
+  const inFolderPath: string = tempFolder;
+  const outFolderPath: string = `${buildFolderPath}/etc`;
+  await FileSystem.ensureFolderAsync(outFolderPath);
+
+  const inFolderPaths: AsyncIterable<string> = enumerateFolderPaths(inFolderPath, '');
+  const outFolderPaths: AsyncIterable<string> = enumerateFolderPaths(outFolderPath, '');
+  const outFolderPathsSet: Set<string> = new Set<string>();
+
+  for await (const outFolderPath of outFolderPaths) {
+    outFolderPathsSet.add(outFolderPath);
+  }
+
+  const nonMatchingFiles: string[] = [];
+  await Async.forEachAsync(
+    inFolderPaths,
+    async (folderItemPath: string) => {
+      outFolderPathsSet.delete(folderItemPath);
+
+      const sourceFileContents: string = await FileSystem.readFileAsync(inFolderPath + folderItemPath);
+      const outFilePath: string = outFolderPath + folderItemPath;
+
+      let outFileContents: string | undefined;
+      try {
+        outFileContents = await FileSystem.readFileAsync(outFilePath);
+      } catch (e) {
+        if (!FileSystem.isNotExistError(e)) {
+          throw e;
+        }
+      }
+
+      const normalizedSourceFileContents: string = Text.convertToLf(sourceFileContents);
+      const normalizedOutFileContents: string | undefined = outFileContents
+        ? Text.convertToLf(outFileContents)
+        : undefined;
+
+      if (normalizedSourceFileContents !== normalizedOutFileContents) {
+        nonMatchingFiles.push(outFilePath);
+        if (!production) {
+          await FileSystem.writeFileAsync(outFilePath, normalizedSourceFileContents, {
+            ensureFolderExists: true
+          });
+        }
+      }
+    },
+    {
+      concurrency: 10
```

Comment:
Although `concurrency: 10` may speed up the build, it would be fairly confusing when trying to investigate a test that is failing.  Consider a design where `--debug` forces `concurrency: 1` (or do we really need concurrency at all?)

Reply:
Are you planning to put the Git repository integration tests in this Rush project as well?  Those probably cannot run in parallel.